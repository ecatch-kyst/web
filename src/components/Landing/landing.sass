
.landing-blob
  position: fixed
  z-index: -1
  transform: scale(2.5)
  display: none
  pointer-events: none
  user-select: none
  opacity: .4
  
  @media (min-width: 1280px)
    display: block

.landing-blob-1
  right: 0
  top: -50px

.landing-blob-2
  left: 0
  bottom: -50px
  transform: scale(2)
  opacity: .2