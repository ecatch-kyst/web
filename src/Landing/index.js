import React, {Component} from 'react'
import {Link, Redirect} from "react-router-dom"
import Centered from '../components/Centered'
import {Typography, Card, Button, CardActions, CardContent, TextField, Grid, Link as MuiLink} from '@material-ui/core'
import {routes} from '../lib/router'
import "./landing.sass"
import blob from "../assets/blob.svg"
import {withNamespaces} from 'react-i18next'
import Store from '../db'


class Landing extends Component {

  static contextType = Store

  state = {
    email: "",
    password: ""
  }

  handleChange = ({target: {name, value}}) => this.setState({[name]: value})

  handleUserLogin = () => {
    const {email, password} = this.state
    this.context.handleUserLogin(email, password)
  }

  render() {
    const {t} = this.props
    const {user} = this.context

    return (
      <Centered>
        {Object.keys(user).length ? <Redirect to={routes.DASHBOARD} /> : null}
        <img alt={t("landing.blob-img-alt")} className="landing-blob landing-blob-1" src={blob}/>
        <img alt={t("landing.blob-img-alt")} className="landing-blob landing-blob-2" src={blob}/>
        <Typography variant="h4">{process.env.REACT_APP_TITLE}</Typography>
        <Card style={{minWidth: "calc(100vw/3)", maxWidth: "calc(100vw - 32px)", margin: 16}}>
          <CardContent>
            <Grid container justify="center" spacing={8}>
              <Grid item>
                <TextField
                  label={t("landing.email")}
                  name="email"
                  onChange={this.handleChange}
                  type="email"
                  variant="outlined"
                />
              </Grid>
              <Grid item>
                <TextField
                  label={t("landing.password")}
                  name="password"
                  onChange={this.handleChange}
                  type="password"
                  variant="outlined"
                />
              </Grid>
            </Grid>
          </CardContent>
          <CardActions>
            <Grid alignItems="center" container justify="flex-end" spacing={16}>
              <Grid item>
                <MuiLink
                  component={Link}
                  style={{marginRight: 16}}
                  to={routes.FORGOT_PASSWORD}
                >{t("landing.forgot-password")}
                </MuiLink>
              </Grid>
              <Grid container item justify="space-between" spacing={8}>
                <Grid item>
                  <Button
                    color="primary"
                    onClick={this.handleUserLogin}
                    size="large"
                    variant="contained"
                  >
                    {t("landing.login")}
                  </Button>
                </Grid>
                <Grid item>
                  <Button
                    color="secondary"
                    component={Link}
                    size="large"
                    to={routes.REGISTER}
                  >
                    {t("landing.register")}
                  </Button>
                </Grid>
              </Grid>
            </Grid>
          </CardActions>
        </Card>
        <Typography variant="caption">{process.env.REACT_APP_VERSION}</Typography>
      </Centered>
    )
  }
}

export default withNamespaces("pages")(Landing)