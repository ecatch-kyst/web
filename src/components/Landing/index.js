import React, {Component} from 'react'
import {Link, Redirect} from "react-router-dom"
import Centered from '../Centered'
import {Typography, Card, Button, CardActions, CardContent, TextField, Grid, Link as MuiLink} from '@material-ui/core'
import {routes} from '../../lib/router'
import "./landing.sass"
import blob from "../../assets/blob.svg"
import {withTranslation} from 'react-i18next'
import Store from '../../db'
import {AUTH} from '../../lib/firebase'


class Landing extends Component {

  static contextType = Store

  state = {
    email: "",
    password: ""
  }

  handleChange = ({target: {name, value}}) => this.setState({[name]: value})

  handleSubmit = e => {
    e.preventDefault && e.preventDefault()
    this.handleUserLogin()
  }

  handleUserLogin = () => {
    const {email, password} = this.state
    this.context.handleUserLogin({email, password})
  }

  render() {
    const {t} = this.props

    return (
      <form onSubmit={this.handleSubmit}>
        <Centered>
          {AUTH.currentUser ? <Redirect to={routes.DASHBOARD} /> : null}
          <img alt={t("landing.blob-img-alt")} className="landing-blob landing-blob-1" src={blob}/>
          <img alt={t("landing.blob-img-alt")} className="landing-blob landing-blob-2" src={blob}/>
          <Typography variant="h4">{process.env.REACT_APP_TITLE}</Typography>
          <Card style={{minWidth: "calc(100vw/3)", maxWidth: "calc(100vw - 32px)", margin: 16}}>
            <CardContent>
              <Grid container direction="column" justify="center" spacing={8}>
                <Grid item>
                  <TextField
                    autoFocus
                    fullWidth
                    label={t("landing.email")}
                    name="email"
                    onChange={this.handleChange}
                    type="email"
                    variant="outlined"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    fullWidth
                    label={t("landing.password")}
                    name="password"
                    onChange={this.handleChange}
                    type="password"
                    variant="outlined"
                  />
                </Grid>
              </Grid>
            </CardContent>
            <CardActions>
              <Grid alignItems="center" container justify="flex-end" spacing={16}>
                <Grid alignItems="center" container item justify="space-evenly" spacing={8}>
                  <Grid item>
                    <Button
                      color="secondary"
                      onClick={this.handleUserLogin}
                      size="large"
                      type="submit"
                      variant="contained"
                    >
                      {t("landing.login")}
                    </Button>
                  </Grid>
                  <Grid item>
                    <MuiLink
                      component={Link}
                      style={{marginRight: 16}}
                      to={routes.FORGOT_PASSWORD}
                    >{t("landing.forgot-password")}
                    </MuiLink>
                  </Grid>
                  <Grid item>
                    <Button
                      color="primary"
                      component={Link}
                      size="large"
                      to={routes.REGISTER}
                    >
                      {t("landing.register")}
                    </Button>
                  </Grid>
                </Grid>
              </Grid>
            </CardActions>
          </Card>
        </Centered>
      </form>
    )
  }
}

export default withTranslation("pages")(Landing)