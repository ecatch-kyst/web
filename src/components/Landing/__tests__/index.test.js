import Landing from ".."

describe("Landing component", () => {
  const wrapper = shallow(<Landing/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

})