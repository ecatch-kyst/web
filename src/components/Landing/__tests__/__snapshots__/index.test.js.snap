// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "i18n": Object {
        "changeLanguage": Object {
          "addMatchers": [Function],
          "advanceTimersByTime": [Function],
          "autoMockOff": [Function],
          "autoMockOn": [Function],
          "clearAllMocks": [Function],
          "clearAllTimers": [Function],
          "deepUnmock": [Function],
          "disableAutomock": [Function],
          "doMock": [Function],
          "dontMock": [Function],
          "enableAutomock": [Function],
          "fn": [Function],
          "genMockFromModule": [Function],
          "isMockFunction": [Function],
          "mock": [Function],
          "requireActual": [Function],
          "requireMock": [Function],
          "resetAllMocks": [Function],
          "resetModuleRegistry": [Function],
          "resetModules": [Function],
          "restoreAllMocks": [Function],
          "retryTimes": [Function],
          "runAllImmediates": [Function],
          "runAllTicks": [Function],
          "runAllTimers": [Function],
          "runOnlyPendingTimers": [Function],
          "runTimersToTime": [Function],
          "setMock": [Function],
          "setTimeout": [Function],
          "spyOn": [Function],
          "unmock": [Function],
          "useFakeTimers": [Function],
          "useRealTimers": [Function],
        },
        "language": "en",
      },
      "t": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "i18n": Object {
          "changeLanguage": Object {
            "addMatchers": [Function],
            "advanceTimersByTime": [Function],
            "autoMockOff": [Function],
            "autoMockOn": [Function],
            "clearAllMocks": [Function],
            "clearAllTimers": [Function],
            "deepUnmock": [Function],
            "disableAutomock": [Function],
            "doMock": [Function],
            "dontMock": [Function],
            "enableAutomock": [Function],
            "fn": [Function],
            "genMockFromModule": [Function],
            "isMockFunction": [Function],
            "mock": [Function],
            "requireActual": [Function],
            "requireMock": [Function],
            "resetAllMocks": [Function],
            "resetModuleRegistry": [Function],
            "resetModules": [Function],
            "restoreAllMocks": [Function],
            "retryTimes": [Function],
            "runAllImmediates": [Function],
            "runAllTicks": [Function],
            "runAllTimers": [Function],
            "runOnlyPendingTimers": [Function],
            "runTimersToTime": [Function],
            "setMock": [Function],
            "setTimeout": [Function],
            "spyOn": [Function],
            "unmock": [Function],
            "useFakeTimers": [Function],
            "useRealTimers": [Function],
          },
          "language": "en",
        },
        "t": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
